GotExplorer.API/Middleware/GlobalExceptionHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿using GotExplorer.BLL.Exceptions;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.WebUtilities;
+using System.Collections;
+
+namespace GotExplorer.API.Middleware
+{
+    public sealed class GlobalExceptionHandler : IExceptionHandler
+    {
+        private readonly ILogger _logger;
+
+        public GlobalExceptionHandler(ILogger<GlobalExceptionHandler> logger)
+        {
+            _logger = logger;
+        }
+
+        public async ValueTask<bool> TryHandleAsync(
+            HttpContext httpContext,
+            Exception exception,
+            CancellationToken cancellationToken)
+        {
+            int status = exception switch
+            {
+                HttpException ex => ex.StatusCode,
+                _ => StatusCodes.Status500InternalServerError,
+            };
```

Comment:
Why do we need this switch at all, we need just map the status codes to response.